hack/lib/util.sh

```diff
@@ -139,6 +135,23 @@ kube::util::gen-doc() {
   ${cmd} "${tmpdir}"
   # create the list of generated files
   ls "${tmpdir}" | LC_ALL=C sort > "${tmpdir}/.files_generated"
+
+  # remove all old generated file from the destination
+  for file in $(cat "${dest}/.files_generated" 2>/dev/null); do
+    if [[ -e "${tmpdir}/${file}" && ! -z "${skipline}" ]]; then
+      # If RCS format diff has 3 lines, only 1 line changed.
+      local fdiff=$(diff -n "${dest}/${file}" "${tmpdir}/${file}")
+      if [[ $(echo "${fdiff}" | wc -l) == 3 && $(echo "${fdiff}" | grep -e "${skipline}") ]]; then
```

Comment:
What is in `${skipline}`? Is it really supposed to be a general grep regexp? Or is it a fixed string? Is it supposed to match a whole line?

This command is wrong, since the second part will actually evaluate the result of `$(echo "${fdiff}" | grep -e "${skipline}")` and that does not make much sense, I think what you wanted was to see whether the `grep` matched or not.

I'm quite confused by what this is trying to do...